extractor/docker/docker.go

```diff
@@ -98,7 +98,15 @@ func ApplyLayers(layers []types.LayerInfo) types.ImageDetail {
 		}
 
 		for _, pkgInfo := range layer.PackageInfos {
-			nestedMap.SetByString(pkgInfo.FilePath, sep, pkgInfo)
+			var pkgs []types.Package
+			for _, pkg := range pkgInfo.Packages {
+				pkg.LayerID = layer.ID
+				pkgs = append(pkgs, pkg)
+			}
+			nestedMap.SetByString(pkgInfo.FilePath, sep, types.PackageInfo{
+				FilePath: pkgInfo.FilePath,
+				Packages: pkgs,
+			})
```

Comment:
nit: you can also write like the following to save memory and lines

```suggestion
			for i, pkg := range pkgInfo.Packages {
				pkgInfo.Packages[i].LayerID = layer.ID
			}
			nestedMap.SetByString(pkgInfo.FilePath, sep, pkgInfo)
```
But I like your way from a readability perspective.

Reply:
I actually like your proposed way better. Much cleaner. I changed to that.